### ISSUE TITLE
Student's project is throwing a TypeError

### ISSUE THREAD

**Issue by ladyadalovelace** [jmoenig/Snap#2496]
https://snap.berkeley.edu/snap/snap.html#present:Username=8706074&ProjectName=U2L1%20click%20alonzo&editMode&noRun

I cannot for the life of me debug this error. The Dev console isn't showing anything at all (at least on the school computers), I've looked at the XML and it seems OK...I've also force-reloaded Snap and cleared cache and reloaded Snap....I'm out of ideas.

I know that the student's project worked in the last class and he was almost done. I've told him to restart the project but it's a shame that he's lost all of this work. Any solutions?

**Comment by jguille2** [comment]
Hi @ladyadalovelace ,

I've downloaded your project and fix the _format_ problem. Here you are

[U2L1 click alonzoRec.xml.zip](https://github.com/jmoenig/Snap/files/3661894/U2L1.click.alonzoRec.xml.zip)

If this recovered version has elements lost, you can also try to recover a previous state of the project. Do you know the _recover_ feature? If not, check [this](https://github.com/jmoenig/Snap/issues/2224#issuecomment-429108136).

Hi @jmoenig The problem is the same that was reported in [this issue](https://github.com/jmoenig/Snap/issues/2224#issuecomment-429108136). I don't know where is this _serializing_ issue.

Joan

**Comment by ladyadalovelace** [comment]
@jguille2 Thank you Joan! I knew about the recover feature but I didn't use it because that would mean the student would lose all their hard work (they only had 2 saved recovery points.)

That being said, can you write some instructions on how to edit the XML file so that the students can get back in? I have about 4-5 that are encountering this issue.

**Comment by jguille2** [comment]
Yes, but I'm sorry about this issue. If you have more than one, we must take care of this.

The workaround:
- Use the project URL, but changing `#present:` for `#dl:` (means "download"). You will get the project _xml_ file.
- Open that _xml_ file with a plain text editor and you can check where is the format problem.
- If it's the same case, search `<ref id` and `</ref>` and change them for `<list id` and `</list>` 

Any problems, continue this forum thread.

Joan

**Comment by ladyadalovelace** [comment]
Between the other teacher and I, we've had about 15 students have the issue. They are all using Chrome (77). Firefox throws a different error:
![image](https://user-images.githubusercontent.com/20193892/65775335-f8698480-e10d-11e9-8dcd-b5ab19c250fc.png)

I'll check all my other student's files and see what's up with theirs. I believe it may be the same issue since they're all throwing the same error.

**Comment by jguille2** [comment]
Hi @ladyadalovelace ,
Two comments...
- About _recovering_. Don't worry to test all the recovery points! Do it! After the recovery you only _load_ that _recovery point_ in your browser. Nothing changes in the cloud. Then you can do whatever you want: _forgeting_ it (if it is not interesting), _saving_ it (if it is good, overwriting the last (with the error) state), or _saving as_ another project (saving a copy without overwriting).

- And about these 15 projects. Give us some feedback (what you have seen) or if you want, you can share their urls and I will check those _xml files with problems_

Joan
